printing/printer.mli

```diff
@@ -194,7 +194,7 @@ val pr_transparent_state   : TransparentState.t -> Pp.t
     records containing the goal and sigma for, respectively, the new and old proof steps,
     e.g. [{ it = g ; sigma = sigma }].
 *)
-val pr_goal                : ?diffs:bool -> ?og_s:(Goal.goal sigma) -> Goal.goal sigma -> Pp.t
+val pr_goal                : ?diffs:bool -> ?og_s:(Goal.goal * Evd.evar_map) -> Goal.goal sigma -> Pp.t
```

Comment:
I just percolated the changes that required removing the `Old` API. Honestly, this whole API is nonsense since `diffs` and `og_s` should be part of the same inductive argument (there is 4 possible cases but only 2 or 3 are relevant) and we should not even expose this as a goal sigma, but rather some abstract type that is indirectly generated by this data and an environment (currently we pick "the" global one, yuck). To add insult to injury, I think that this is actually dead code, we only use lower-level primitives to perform goal diffing.

Reply:
(Take this as a promise to clean up this API in a later PR.)